### DISCUSSION TITLE
MSRV (Minimum Supported Rust Version) policy?

### DISCUSSION THREAD

**Discussion by Desdaemon** [fzyzcjy/flutter_rust_bridge#588] [Polls]
I don't think we have anything like this yet, but it would be important not only for the consumers of this library but also contributors  like myself who would like to use newer features of rustc. (This is only for the codegen program, the runtime support library should  be able to work with most Rust versions supporting the 2018 edition.)

/cc @fzyzcjy

**Comment by fzyzcjy** [comment]
Good question. I personally do not have a strong preference. What is your choice?

In addition, after we provide prebuilt binary for all platforms, codegen's rust version can be almost arbitrary without affecting users

**Comment by gutenfries** [comment]
Hi folks,

working on a completely different PR, I needed to know our MSRV to ensure compatibility.

In many PRs/issues, we talk about the MSRV, but nowhere is there a magic number or a range.

In accordance with our rust edition set to 2018,
https://github.com/fzyzcjy/flutter_rust_bridge/blob/fe67cb5bebea24f5e8901a7849ee035af9081874/Cargo.toml#L13

I propose that we officially (after testing and CIing, of course) declare `nightly-2018-12-06` as the MSRV.

this is the first available 2018 edition of Rust, i.e. the oldest version the compiler will let you use anyway, without modifications.

Web _requires_ nightly, and, to my knowledge, there may or may not be unstable APIs in other sections of code.

I understand that this discussion is concerning the _codegen_ portion of the codebase, but I believe we can all see the benefits in declaring the same MSRV for the entire codebase, as well as the consumable.

ASIDE FROM testing, let me know if anyone has input on this front.

/cc @fzyzcjy @Desdaemon @rogurotus

**Reply by fzyzcjy** [reply]
(I replied in https://github.com/fzyzcjy/flutter_rust_bridge/issues/1310)
